packages/next/src/views/Dashboard/Modular/client.tsx

```diff
@@ -0,0 +1,170 @@
+'use client'
+
+import type { Widget } from 'payload'
+import type { Layout } from 'react-grid-layout'
+
+import { ItemsDrawer, XIcon } from '@payloadcms/ui'
+import { Button } from '@payloadcms/ui/elements/Button'
+import { DrawerToggler } from '@payloadcms/ui/elements/Drawer'
+import { type Option, ReactSelect } from '@payloadcms/ui/elements/ReactSelect'
+import { SetStepNav } from '@payloadcms/ui/elements/StepNav'
+import React, { useId } from 'react'
+import { Responsive, WidthProvider } from 'react-grid-layout'
+
+import { useDashboardLayout } from './useDashboardLayout.js'
+
+const ResponsiveGridLayout = WidthProvider(Responsive)
+
+const BREAKPOINT = 768
+
+export type WidgetInstanceClient = {
+  clientLayout: Layout
+  component: React.ReactNode
+}
+
+export function ModularDashboardClient({
+  clientLayout: initialLayout,
+  widgets,
+}: {
+  clientLayout: WidgetInstanceClient[]
+  widgets: Widget[]
+}) {
+  const {
+    addWidget,
+    cancel,
+    currentLayout,
+    deleteWidget,
+    handleLayoutChange,
+    isEditing,
+    resetLayout,
+    saveLayout,
+    setIsEditing,
+  } = useDashboardLayout(initialLayout)
+
+  const uuid = useId()
+  const drawerSlug = `widgets-drawer-${uuid}`
+
+  return (
+    <div>
+      <SetStepNav
+        nav={[
+          {
+            label: (
+              <DashboardBreadcrumbDropdown
+                isEditing={isEditing}
+                onCancel={cancel}
+                onEditClick={() => setIsEditing(true)}
+                onResetLayout={resetLayout}
+                onSaveChanges={saveLayout}
+                widgetsDrawerSlug={drawerSlug}
+              />
+            ),
+          },
+        ]}
+      />
+      <ResponsiveGridLayout
+        breakpoints={{ lg: BREAKPOINT, xxs: 0 }}
+        className={`grid-layout ${isEditing ? 'editing' : ''}`}
+        cols={{ lg: 12, xxs: 6 }}
+        isDraggable={isEditing}
+        isResizable={isEditing}
+        layouts={{
+          lg: currentLayout.map((item) => item.clientLayout),
+        }}
+        onLayoutChange={handleLayoutChange}
+        rowHeight={(BREAKPOINT / 12) * 3}
+        useCSSTransforms={false}
+      >
+        {currentLayout &&
+          currentLayout.map((widget) => (
+            <div className="widget" key={widget.clientLayout.i}>
+              <div className={`widget-wrapper ${isEditing ? 'widget-wrapper--editing' : ''}`}>
+                <div className="widget-content">{widget.component}</div>
```

Comment:
Should the widget container have padding or be a concern of the widget component?